packages/data-models/src/lemma.js

```diff
@@ -166,17 +167,80 @@ class Lemma {
   }
 
   /**
-   * Test to see if two lemmas are full homonyms
+   * Test to see if two lemmas are full homonyms.
    *
-   * @param {Lemma} lemma the lemma to compare
-   * @returns {boolean} true or false
+   * @param {Lemma} lemma - the lemma to compare.
+   * @param {object} options - Additional comparison options.
+   * @param {boolean} options.normalize - Whether to normalize words before comparison.
+   * @returns {boolean} true or false.
    */
-  isFullHomonym (lemma) {
-    // returns true if the word and part of speech match
-    return this.word === lemma.word &&
-      this.features[Feature.types.part] &&
-      lemma.features[Feature.types.part] &&
-      this.features[Feature.types.part].isEqual(lemma.features[Feature.types.part])
+  isFullHomonym (lemma, { normalize = false } = {}) {
+    // If parts of speech do not match this is not a full homonym
+    if (!this.features[Feature.types.part] ||
+      !lemma.features[Feature.types.part] ||
+      !this.features[Feature.types.part].isEqual(lemma.features[Feature.types.part])) {
+      return false
+    }
+
+    // Check if words are the same
+    const areSameWords = normalize
+      ? LMF.getLanguageModel(this.languageID).compareWords(this.word, lemma.word, true,
+        { normalizeTrailingDigit: true, normalizeWord: true })
+      : this.word === lemma.word
+
+    return areSameWords
+  }
+
+  /**
+   * Disambiguate between this and the other lemma.
+   *
+   * @param {string} otherLemma - The other lemma for disambiguation.
+   * @returns {string} - A disambiguated word.
+   */
+  disambiguate (otherLemma) {
+    // Use a special logic for Greek words
+    if (this.languageID === Constants.LANG_GREEK) {
```

Comment:
> should there be a test on whether or not these are the same words, once normalized, before proceeding?

Do you mean we should not proceed if those words are not the same? We have a call to `canBeDisambiguatedWith()` (https://github.com/alpheios-project/alpheios-core/blob/i166-lexeme-disambiguation/packages/data-models/src/lexeme.js#L154) where we check if words are the same after normalization. If they are not, we will not proceed with calling a `lexeme.disambiguate()`.

Or do you think we should add that check to a `disambiguate()` method? If so, should there be any special logic associated with it?

Reply:
> I'm not sure why this is restricted to Greek words. it should apply to the other languages too.

I thought this logic makes sense within the context of Greek only.

If not, then, I think, we should create `hasTrailingDigit()`, `needsNormalization()`, and `hasUpperCase()` methods in the base `LanguageModel` class, because those methods will be called for every language then. I'm just not sure about normalization. Will any language has it?

Reply:
> > should there be a test on whether or not these are the same words, once normalized, before proceeding?
> 
> Do you mean we should not proceed if those words are not the same? We have a call to `canBeDisambiguatedWith()` (https://github.com/alpheios-project/alpheios-core/blob/i166-lexeme-disambiguation/packages/data-models/src/lexeme.js#L154) where we check if words are the same after normalization. If they are not, we will not proceed with calling a `lexeme.disambiguate()`.
> 
> Or do you think we should add that check to a `disambiguate()` method? If so, should there be any special logic associated with it?

I am thinking about reuse here. Because the check on whether the words are the same takes place outside of the disambiguate method, it could produce incorrect results if called from elsewhere. So I think it would probably be better to do the check here too, even if it's duplicated in the current workflow.

Reply:
> > I'm not sure why this is restricted to Greek words. it should apply to the other languages too.
> 
> I thought this logic makes sense within the context of Greek only.
> 
> If not, then, I think, we should create `hasTrailingDigit()`, `needsNormalization()`, and `hasUpperCase()` methods in the base `LanguageModel` class, because those methods will be called for every language then. I'm just not sure about normalization. Will any language has it?

No, it's not Greek specific. There is already a normalizeWord method in the base language class (it just returns the word). So yes I think these methods should be added to the base language class with default behavior.

Reply:
Good points, will update the code.